### PR TITLE
Introduce Intel SGX to Multicloud GitOps pattern

### PR DESCRIPTION
**Pull request by polszewska**

### CODE REVIEW

File: content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+.Additional resources
+
+View and download all of the diagrams above from the link:https://www.redhat.com/architect/portfolio/tool/index.html?#gitlab.com/osspa/portfolio-architecture-examples/-/raw/main/diagrams/spi-multi-cloud-gitops.drawio[Red Hat Portfolio Architecture open source tooling site].
+
+[id="extension-of-mcg"]
+== {mcg-sgx-hello-world-pattern}
+
+The basic {mcg-pattern} has been extended to highlight the *{intel-5th-gen-xeon-processors}* capabilities, offering developers {intel-sgx} as a streamlined pathway to protect data in use against threats with isolation, encryption and attestation while also allowing users to maintain control and use their data.
+
+The application which has been added to the basic pattern is named `hello-world-sgx`. It is simple Python application that runs securely inside SGX enclave. SGX enclave is encrypted memory, which prevents unauthorized access from other containers running on the same host or even some host processes.
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc]: The application which has been added to the basic pattern is named `hello-world-sgx`. It is **_a_** simple Python application that runs securely inside **_the_** SGX enclave. SGX enclave is _**an**_ encrypted memory, which prevents unauthorized access from other containers running on the same host or even some **_host's_** processes.

File: content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+The sample application uses image converted with Gramine Shielded Containers (GSC) tools to include Intel SGX related information and execute the application inside an Intel SGX enclave using the Gramine Library OS. It's easy way to convert the application into its secure version.
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc]: **_Intel SGX-related_**

It’s _**an**_ easy way to convert the application into its secure version.

File: content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+
+[role="_additional-resources"]
+.Additional resources
+
+View and download all of the diagrams above from the link:https://www.redhat.com/architect/portfolio/tool/index.html?#gitlab.com/osspa/portfolio-architecture-examples/-/raw/main/diagrams/spi-multi-cloud-gitops.drawio[Red Hat Portfolio Architecture open source tooling site].
+
+[id="extension-of-mcg"]
+== {mcg-sgx-hello-world-pattern}
+
+The basic {mcg-pattern} has been extended to highlight the *{intel-5th-gen-xeon-processors}* capabilities, offering developers {intel-sgx} as a streamlined pathway to protect data in use against threats with isolation, encryption and attestation while also allowing users to maintain control and use their data.
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc]: encryption, **_[comma]_** and attestation

File: content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+
+The sample application uses image converted with Gramine Shielded Containers (GSC) tools to include Intel SGX related information and execute the application inside an Intel SGX enclave using the Gramine Library OS. It's easy way to convert the application into its secure version.
+
+Since the `hello-world-sgx` application must be running on the node with CPU supporting {intel-sgx}, Node Feature Discovery Operator (NFD) and Intel Device Plugins Operator (IDP) are deployed as a part of this pattern.
+
+NFD manages the detection of hardware features and configuration in an OpenShift Container Platform cluster. It labels the nodes with hardware-specific information. It is a requirement for IDP installation. IDP provides resources which are the user interface to claim and consume the hardware feature by user pods and also advertises Intel-specific hardware resources to the kubelet.
+
+In the logs of `hello-world-sgx` pod there is information that "Gramine is starting" and it executes the application by printing "HelloWorld!".
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc]: In the logs of `hello-world-sgx` pod, **_[comma]_** there is...

File: content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+
+This pattern demonstrates basic capabilities of running docker applications inside SGX enclaves. Based on this pattern other applications can be secured with {intel-sgx} and Gramine Shielded Containers in similar way as presented here.
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc]: ... Containers in **_a_** similar way as presented here.

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+== Deploying the {mcg-sgx-hello-world-pattern}
+
+.Prerequisites
+
+* An OpenShift cluster
+ ** To create an OpenShift cluster, go to the https://console.redhat.com/[Red Hat Hybrid Cloud console] and select *Services \-> Containers \-> Create cluster*.
+ ** The cluster must have a dynamic `StorageClass` to provision `PersistentVolumes`.
+ ** link:../../multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-cluster-sizing[Cluster sizing] requirements.
+* Optional: A second OpenShift cluster for multicloud demonstration.
+* {intel-sgx} enabled in BIOS (Basic Input Output System) on at least one worker machines - specific steps depends on platform.
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: Intel SGX **_must be_** enabled in BIOS

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+ ** Install the https://gramine.readthedocs.io/projects/gsc/en/latest/#prerequisites[GSC prerequisites].
+
+The use of this pattern depends on having at least one running Red Hat OpenShift cluster. However, consider creating a cluster for deploying the GitOps management hub assets and a separate cluster for the managed cluster.
+
+If you do not have a running Red Hat OpenShift cluster, you can start one on a
+public or private cloud by using https://console.redhat.com/openshift/create[Red Hat Hybrid Cloud Console].
+
+.Procedure
+
+. https://validatedpatterns.io/learn/quickstart/[Install the tooling dependencies].
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: It is doubled

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+----
+git push origin my-branch
+----
+
+. Prepare and build `hello-world` docker image and convert it to secure version using GSC - described below.
+
+. Deploy the pattern by running `./pattern.sh make install` or by using the link:/infrastructure/using-validated-pattern-operator/[Validated Patterns Operator] - both methods are described below.
+
+[id="build-docker-image-and-convert-to-sgx"]
+== Prepare docker image and convert it to use Intel SGX
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: 1. {intel-sgx}
2. please add `Optional: ` at the beginning of the title

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc [hunk 9]
```diff
@@ -0,0 +1,325 @@
...
+. Copy configuration file from sample template:
++
+[source,terminal]
+----
+$ cd gsc
+$ cp config.yaml.template config.yaml
+----
+(Optional) This file can be adjusted if needed.
+
+. Create new file _hello-world.manifest_ containing parameters of conversion:
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: Create **_a new hello-world.manifest file_** containing...

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc [hunk 10]
```diff
@@ -0,0 +1,325 @@
...
+----
+$ ./gsc build hello-world hello-world.manifest
+----
+. Sign the graminized docker image:
++
+[source,terminal]
+----
+$ ./gsc sign-image hello-world enclave-key.pem
+----
+. Push secured container image to registry which is available on the cluster:
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: Push secured container image to **_the_** registry which is available on the cluster:

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc [hunk 11]
```diff
@@ -0,0 +1,325 @@
...
+$ ./gsc build hello-world hello-world.manifest
+----
+. Sign the graminized docker image:
++
+[source,terminal]
+----
+$ ./gsc sign-image hello-world enclave-key.pem
+----
+. Push secured container image to registry which is available on the cluster:  
+.. Tag secured container image with appropriate tag and push the image to a registry:
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: Tag secured container image with appropriate tag and push **_it to the_** registry:

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc [hunk 12]
```diff
@@ -0,0 +1,325 @@
...
+.Install Validated Patterns Operator
+image::multicloud-gitops-sgx-hello-world/sgx-validated-pattern-operator.png[Install Validated Patterns Operator,scale=50]
++
+. Choose default settings for the installation mode, namespaces and update strategy and confirm it by clicking *Install*.
+
+. Select *Operators > Installed Operators*.
+
+. Ensure that *Validated Patterns Operator* is listed in the `openshift-operators` project with a status `Succeeded`.
+
+After succeeded installation open *Validated Patterns Operator*, go to *Pattern* tab and click *Create Pattern*.
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: After a successful installation open the *Validated Patterns Operator* page. Next, go to *Pattern* tab and click *Create Pattern*.

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc [hunk 13]
```diff
@@ -0,0 +1,325 @@
...
+
+. Fill the *Name* of the pattern `multicloud-gitops-sgx-hello-world` and *Cluster Group Name* `hub` (from _values-global.yaml_ file).
+
+. Under *Git Config > Target Repo* copy the link to your fork and under *Git Config > Target Revision* write the name of your branch (Figure 3).
+
+. Click *Create* to create the pattern.
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: It is a bit hard to understand. I get it, but only thanks to the image.
Can we rephrase it? Maybe sth like:

1. Set the _Name_ field to `multicloud-gitops-amx-rhoai`, and the _Cluster Group Name_ to `hub`. Values must be the same as in the _values-global.yaml_ file (Figure 3).
2. As a *Git Config > Target Repo* value, paste the link to your fork. Under *Git Config > Target Revision* write the name of your branch (Figure 3).
3. Click the _Create_ button to create the pattern.

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc [hunk 14]
```diff
@@ -0,0 +1,325 @@
...
+----
++
+
+. After login go to `secret` catalog and clik *Create secret* and fill all the fields manually (Figure 2):
+
+.. *Path for this secret* is `global/config-demo` (from _values.yaml_ file for `config-demo` charts)
+
+.. Under *Secret data* key is `secret` (from _values-secret-multicloud-gitops.yaml_ file) and in next field put its value.
+
+.. Click *Add* and then *Save*.
```
Comment (pgrabusz) [inline content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: TBH I don't get it... Let me try to rephrase it, and decide if this is clear:

a) Put `global/config-demo` value in the _Path for this secret_ field (the value cames from _values-secret-multicloud-gitops.yaml_ file).
b) Add one _Secret data_ key-value pair. Put `secret` as a key (left field). The value must acquired from _values-secret-multicloud-gitops.yaml_ file. Click _Add_ button to confirm.
c) Click _Save_ to save changes.

THIS IS HOW I UNDERSTAND THIS PART - PLEASE VALIDATE A LOGIC BEFORE REPLACING YOUR PART ;)

**BTW is this section twice? In the "Getting started" page, and in the "Demo Script"?**

**Review by pgrabusz — CHANGES_REQUESTED** [review]
Great job Paulina :)
small comments in-line + here's the cluster sizing we should include in the "Cluster sizing" section. You can copy-paste it from Tomasz or Iza when they change it.

In the Cluster sizing tab, please change recommended cluster size:
* Control Planes CPU: 2x 5th Gen Intel® Xeon® Gold 6526Y (16 cores at 2.8 GHz base with Intel AXM) or better
* Workers CPU: 2x 5th Gen Intel® Xeon® Gold 6538Y+(32 cores at 2.2 GHz base with Intel AXM) or better

In the Cluster sizing tab, please change the "tested in the on-premises environment with following HW" table to:

* CP + W CPU: 2x 5th Gen Intel® Xeon® Platinum 8568Y+ (48 cores at 2.3 GHz base with Intel AXM)
* CP + W Memory: 512 GB (16x32GB DDR5 5600)

File: content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc (hunk 1 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc]: Updated

File: content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc (hunk 2 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc]: Updated

File: content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc (hunk 3 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc]: Updated

File: content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc (hunk 4 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc]: Updated

File: content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc (hunk 5 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/_index.adoc]: Updated

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc (hunk 6 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: Updated

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc (hunk 7 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: Removed from this section

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc (hunk 8 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: Updated

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc (hunk 9 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: Updated

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc (hunk 10 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: Updated

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc (hunk 11 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: Updated

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc (hunk 12 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: Updated

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc (hunk 13 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: Updated

File: content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc (hunk 14 above)
Comment (polszewska) [reply content/patterns/multicloud-gitops-sgx-hello-world/mcg-sgx-hello-world-getting-started.adoc]: Checked and corrected. Left this part only in "getting started" section.

**Pull request merged**